concordium-node/src/stats_export_service.rs

```diff
@@ -67,11 +72,84 @@ impl prometheus::core::Collector for GrpcInFlightRequestsCollector {
     }
 }
 
+/// Collector tracking stats which needs to be synced with consensus on scrape
+/// through the FFI.
+pub struct StatsConsensusCollector {
+    /// Reference to consensus to support consensus queries.
+    consensus:              ConsensusContainer,
+    /// The baking committee status of the node for the current best block.
+    baking_committee:       IntGaugeVec,
+    /// Whether the node is a member of the finalization committee for the
+    /// current best block.
+    finalization_committee: IntGauge,
+}
+
+impl StatsConsensusCollector {
+    pub fn new(consensus: ConsensusContainer) -> anyhow::Result<Self> {
+        let baking_committee = IntGaugeVec::new(
+            Opts::new(
+                "consensus_baking_committee",
+                "The baking committee status of the node for the current best block, labelled \
+                 with the status where the only metric with value of 1 represents the current \
+                 status",
+            )
+            .variable_label("status"),
+            &["status"],
+        )?;
+
+        let finalization_committee = IntGauge::with_opts(Opts::new(
+            "consensus_finalization_committee",
+            "The finalization commmittee status of the node for the current best block. The \
+             metric will have a value of 1 when member of the finalization committee",
+        ))?;
+
+        Ok(Self {
+            consensus,
+            baking_committee,
+            finalization_committee,
+        })
+    }
+
+    /// Update labelled metric to reflect the provided baking committee status.
+    fn set_baking_committe(&self, status: ConsensusIsInBakingCommitteeResponse) {
+        for status_label in ConsensusIsInBakingCommitteeResponse::labels() {
+            self.baking_committee.with_label_values(&[status_label]).set(0);
+        }
+        self.baking_committee.with_label_values(&[status.label()]).set(1);
+    }
```

Comment:
This seems rather awkward. Why did you not choose to instead have a gauge and just set it to the current value?

That seems lighter.

Reply:
Do you mean map the enum to a number and set the gauge to this number?

Reply:
Yes.

Reply:
For others reading, this was discussed offline and the conclusion was to have one metric, where the value represents the variant as suggested by Ales.
This avoids updating multi metrics and makes the code simpler. We did not find any major downsides to doing this.